store/dataStore.ts

```diff
@@ -1,17 +1,33 @@
 import { create } from 'zustand';
+import { produce } from 'immer';
+import { IUser } from '@/api/IapiFunctions';
 
-interface Example {
-    id: number;
-    name: string;
+interface State {
+  user: IUser;
 }
-interface ExampleStore {
-    examples: Example[];
+
+interface Action {
+  updateUser: (id: State['user']['id'], email: State['user']['email']) => void;
```

Comment:
I didn't see this earlier, but you should be able to do something like this. 
```suggestion
  updateUser: ({id , email}: IUser['user']) => void;
```

Reply:
✅ Done per our Discord Chat